pkg/component/controller/cplb_unix.go

```diff
@@ -0,0 +1,328 @@
+//go:build unix
+
+/*
+Copyright 2024 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net"
+	"os"
+	"path/filepath"
+	"slices"
+	"text/template"
+
+	"github.com/k0sproject/k0s/internal/pkg/dir"
+	"github.com/k0sproject/k0s/internal/pkg/users"
+	k0sAPI "github.com/k0sproject/k0s/pkg/apis/k0s/v1beta1"
+	"github.com/k0sproject/k0s/pkg/assets"
+	"github.com/k0sproject/k0s/pkg/config"
+	"github.com/k0sproject/k0s/pkg/constant"
+	"github.com/k0sproject/k0s/pkg/supervisor"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+// Keepalived is the controller for the keepalived process in the control plane load balancing
+type Keepalived struct {
+	K0sVars    *config.CfgVars
+	Config     *k0sAPI.ControlPlaneLoadBalancingSpec
+	uid        int
+	supervisor *supervisor.Supervisor
+	log        *logrus.Entry
+}
+
+// Init extracts the needed binaries and creates the directories
+func (k *Keepalived) Init(_ context.Context) error {
+	if k.Config == nil {
+		return nil
+	}
+	k.log = logrus.WithField("component", "CPLB")
+
+	var err error
+	k.uid, err = users.GetUID(constant.KeepalivedUser)
+	if err != nil {
+		k.log.Errorf("running keepalived as root: %v", err)
```

Comment:
I don't think this is an error per se, maybe a `Warn` level?

Reply:
It should. Fixed.